internal/webhook/sparkapplication_defaulter_test.go

```diff
@@ -0,0 +1,215 @@
+/*
```

Comment:
I think the `sparkapplication_defauler_test.go` file should be used to implement unit tests against funcs related to `SparkApplicationDefaulter`, such as `defaultSparkApplication`. We do not need to instantiate the webhook manually and call the handle func. Maybe we can remove this file in this PR and later add a new e2e test in another PR to validate that the serialize/deserialize process can work as expected.

Reply:
I created an issue so ths doesn't get forgotten https://github.com/kubeflow/spark-operator/issues/2284